engine/inputdata/blueprint.ftl

```diff
@@ -21,6 +21,238 @@
     [/#if]
 [/#macro]
 
+[#function getBlueprintAttributes ]
+    [#local attributes = []]
+
+    [#-- Layers --]
+    [#local layerChildren = []]
+    [#list layerConfiguration as layer,layerConfig]
+        [#local layerChildren += [
+            {
+                "Names" : layerConfig.ReferenceLookupType,
+                "SubObjects" : true,
+                "Children" : layerConfig.Attributes
+            },
+            {
+                "Names" : layer,
+                "Children" : layerConfig.Attributes
+            }
+        ]]
+    [/#list]
+
+    [#-- Components --]
+    [@includeAllComponentDefinitionConfiguration
+        SHARED_PROVIDER
+        commandLineOptions.Deployment.Provider.Names
+    /]
+
+    [#local componentChildren = []]
+    [#local subComponentNames = 
+        asFlattenedArray(
+            componentConfiguration
+            ?keys
+            ?filter(c -> componentConfiguration[c].Components??)
+            ?map(c -> componentConfiguration[c].Components)
+        )?map(c -> c.Component)]
+
+    [#list componentConfiguration as component,config]
+
+        [#-- Skip any subComponents --]
+        [#if !(subComponentNames?seq_contains(component))]
+            [#local attrs = asFlattenedArray(config["ResourceGroups"]["default"].Attributes?values) + coreComponentChildConfiguration ]
+
+            [#-- Include subComponents as valid Attribute children --]
+            [#local components = []]
+            [#if config.Components??]
+                [#list config.Components as child]
+                    [#local components += [{
+                        "Names" : child.Component,
+                        "SubObjects" : true,
+                        "Children" : asFlattenedArray(componentConfiguration[child.Type]["ResourceGroups"]["default"].Attributes?values) + coreComponentChildConfiguration
+                    }] ]
+                [/#list]
+            [/#if]
```

Comment:
I think any resource group can provide configuration so might be better to loop over all potential resource groups ( currently there is only the default one )

Reply:
Updated here: 51ca5d4